### ISSUE TITLE
[BUG] (Secondary Serial Port Not Functional On MSK_GEN_L_V2 Board)

### ISSUE THREAD

**Issue by borland1** [MarlinFirmware/Marlin#17436]
### Bug Description
Secondary Serial Port is not functioning.  Serial communication exists only on primary SERIAL_PORT. 

When connecting FTDI serial converter (TX, RX, Gnd) between PC and MEGA 2560 Serial1 at RAMPS Z-MIN and Z-MAX board pins, no serial communication exists.

### My Configurations
Board is MKS_GEN_L_V2

Configuration.h file:
#define SERIAL_PORT 0
#define SERIAL_PORT_2 1

#define Z_MIN_PIN 15  // remapping Z_MIN_PIN function to otherwise unused Y_MAX_PIN (Pin 15)
#define Z_MAX_PIN -1 // disable Z-MAX pin function

[2.0.5.3ConfigFiles.zip](https://github.com/MarlinFirmware/Marlin/files/4447376/2.0.5.3ConfigFiles.zip)

**Comment by borland1** [comment]
I was comparing schematics for boards MKS_GEN_L_V2, MKS_GEN_V1, and MKS_SGEN_L.  These boards all have pull-up resistors on all End-Stop Connectors (X-, X+, Y-, Y+, Z-, Z+).  Each End-Stop Signal input pin has one decoupling capacitor to ground and also dropping resistor between Signal and it's respective processor pin.

On board MKS_SGEN_L (LPC1768 processor), the secondary serial port is available on AUX-1. That's not the case with MKS_GEN_V1 or MKS_GEN_V2 which have AUX-1 wired to the same pins as the onboard USB port.

On MKS_GEN_V1, the value of each pull-up resistor is 4.7K ohm and the dropping resistor is 1K ohm. They are also very small surface mounted resistors ( 0603 SMD ).

I have a Arduino Mega development board that I can try out to see if I can get SERIAL_PORT_2 communicting with the PC.  If that works, I will then try de-soldering resistors R3 ( Z- ) and R5 ( Z+) pull-up resistors and retest secondary serial port ( SERIAL_PORT_2 ).

**Comment by borland1** [comment]
Great news.  I uploaded same 2.0.5.3 firmware configuration to my Arduino Mega development board, after changing two Configuration.h lines..

```
#define TEMP_SENSOR_0 998   // dummy always reads 25 degrees C
#define TEMP_SENSOR_BED 998 // dummy always reads 25 degrees C
```

I will now try de-soldering pull-up resistors  R3 ( Z- ) and R5 ( Z+) on MKS_GEN_L_V2 board, then retest SERIAL_PORT_2.

I did notice this behaviour....  Nothing appear on the secondary serial monitor, unless I manually send g-code commands.  Anyone know if the boot-up echo messages, that appear on the primary serial ( SERIAL_PORT ), can be turned ON for the secondary serial port ( SERIAL_PORT_2 ) ?

**Comment by borland1** [comment]
Problem is resolved.

I removed surface mounted pull-up resistors R3 and R4, but that did not enable serial1 communications.

I then removed the surface mounted decoupling capacitors C2 and C3.  They are also very small size ( 0603 SMD ).  After removing R3, R4, C2 and C3, I was able to enable secondary serial port SERIAL_PORT_2.

There is are still two 1K ohm resistors ( R4 and R6 ) between the Signal pins ( Z-, Z+ ) and their associated processor pins.  From my limited testing, this does not appear to affect serial communications at 250,000 baud rate.  Ideally, these resistors could be replaced with 0 ohm surface mounted ( 0603 SMD ) resistors, but it is much easier to just leave them as-is.
